compiler/rustc_parse/src/lexer/tokentrees.rs

```diff
@@ -186,21 +180,21 @@ impl<'psess, 'src> Lexer<'psess, 'src> {
     // Cut-down version of `bump` used when the token kind is known in advance.
     fn bump_minimal(&mut self) -> Spacing {
         let (next_tok, is_next_tok_preceded_by_whitespace) = self.next_token_from_cursor();
+        let this_spacing = self.calculate_spacing(&next_tok, is_next_tok_preceded_by_whitespace);
+        self.token = next_tok;
+        this_spacing
+    }
 
-        let this_spacing = if is_next_tok_preceded_by_whitespace {
+    fn calculate_spacing(&self, next_tok: &Token, is_preceded_by_whitespace: bool) -> Spacing {
```

Comment:
the bool param should not be used as only 1/2 callsites need it - so move it to that one